transcom/mymove
# B-21363 Prime Updates INT

## [Agility ticket](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-21363)
## [First B-21363 PR (combined with B-21362)](https://github.com/transcom/mymove/pull/13837)
## [Second B-21363 PR](https://github.com/transcom/mymove/pull/13933)

## Summary

Adding unaccompanied_baggage FF checks to the prime API's createMTOShipment endpoints (v2 and v3, even though I think only v3 is used)

To test:
Example body for creating a UB shipment:
```
{
  "moveTaskOrderID": "91154a53-6880-460b-a270-57bcb53d631a",
  "shipmentType": "UNACCOMPANIED_BAGGAGE",
  "requestedPickupDate": "2024-10-04",
  "pickupAddress": {
      "streetAddress1": "123 Main St",
      "city": "Nicholasville",
      "state": "KY",
      "postalCode": "40356",
      "country": "US",
      "county": "JESSAMINE"
   }
}
```

v2, FF false
1. Make sure unaccompanied_baggage feature flag is false in your .envrc.
2. Call the v2 createMTOShipment: `http://primelocal:3000/prime/v2/mto-shipments`
3. Verify that you receive a 422 response with a message of "Unaccompanied baggage shipments can't be created unless the unaccompanied_baggage feature flag is enabled"

v3, FF false
1. Make sure unaccompanied_baggage feature flag is false in your .envrc.
2. Call the v2 createMTOShipment: `http://primelocal:3000/prime/v3/mto-shipments`
3. Verify that you receive a 422 response with a message of "Unaccompanied baggage shipments can't be created unless the unaccompanied_baggage feature flag is enabled"

v2, FF true
1. Make sure unaccompanied_baggage feature flag is true in your .envrc.
2. Call the v2 createMTOShipment: `http://primelocal:3000/prime/v2/mto-shipments`
3. Verify that you receive a 200 and can create the UB shipment

v3, FF true
1. Make sure unaccompanied_baggage feature flag is true in your .envrc.
2. Call the v2 createMTOShipment: `http://primelocal:3000/prime/v3/mto-shipments`
3. Verify that you receive a 200 and can create the UB shipment
